### PR TITLE
Expose signature timestamps

### PR DESCRIPTION
**Pull request by alxdarksage**
For Bridge PR #1834.

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java [hunk 1]
```diff
@@ -319,6 +329,7 @@ private static void giveAndGetConsentHelper(String name, LocalDate birthdate, St
             assertEquals("birthdate matches", birthdate, sigFromServer.getBirthdate());
             assertEquals("imageData matches", imageData, sigFromServer.getImageData());
             assertEquals("imageMimeType matches", imageMimeType, sigFromServer.getImageMimeType());
+            assertNotNull(sigFromServer.getSignedOn());
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java]: Should we also check withdrewOn after withdrawing?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java]: Added a check (researcher account retrieves the participant and verifies the history object for this signature has a withdrawn timestamp).

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java [hunk 2]
```diff
@@ -350,6 +362,15 @@ private static void giveAndGetConsentHelper(String name, LocalDate birthdate, St
...
             assertFalse(status.isSignedMostRecentConsent());
+            assertNull(status.getSignedOn());
+            
+            // Get the consent signature and verify it is withdrawn. You can't get it as the test 
+            // user... the user is withdrawn! 
+            ParticipantsApi participantsApi = researchUser.getClient(ParticipantsApi.class);
+            StudyParticipant retrieved = participantsApi.getParticipantById(testUser.getUserId(), true).execute().body();
+            
+            List<UserConsentHistory> history = retrieved.getConsentHistories().get(testUser.getDefaultSubpopulation());
+            assertTrue( history.get(0).getWithdrewOn().isAfter(DateTime.now().minusMinutes(2)) );
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java]: Jenkins clock skew can be as high as 10 minutes. This test may fail sporadically.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java [hunk 3]
```diff
@@ -2,27 +2,71 @@
...
+    @Test
+    public void verifySession() throws Exception {
+        DateTime startOfTest = DateTime.now();
+        
+        TestUser user = TestUserHelper.createAndSignInUser(SessionTest.class, true);
+        
+        UserSessionInfo session = user.getSession();
+        assertNotNull(session.getId());
+        assertEquals(SharingScope.NO_SHARING, session.getSharingScope());
+        assertTrue(session.getCreatedOn().isAfter(startOfTest));
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SessionTest.java]: Similarly, this is likely to fail sporadically (and again later in this same test).

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**